x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/lib/is_rate.ts

```diff
@@ -9,21 +9,30 @@ import { has } from 'lodash';
 import { MetricsUIAggregation } from '../../../../../common/inventory_models/types';
 import { SnapshotCustomMetricInput } from '../../../../../common/http_api';
 
-export const isMetricRate = (metric: MetricsUIAggregation): boolean => {
-  const values = Object.values(metric);
-  return values.some((agg) => has(agg, 'derivative')) && values.some((agg) => has(agg, 'max'));
+export const isMetricRate = (metric: MetricsUIAggregation | undefined): boolean => {
+  if (metric != null) {
```

Comment:
@mgiota, Could you help me understand this check, I found it a bit confusing. 

In the `isMetricRate` function signature TS tells that `metric` is either `MetricsUIAggregation` OR `undefined`. In the function, the check is against the `null` and the check is done with the `!=` operator, and NOT a strict (which I prefer) `!==`
When/Could`metric` be `null`? Shouldn't we stick with one type for the falsy values?

Reply:
`!= null` is a pretty common way to check for `null` and `undefined` at the same time. Checking for `Global.undefined` itself is slightly dangerous because it's not a reserved name and can be reassigned.

Reply:
My point is we still don't know what`metric` could be when it's `falsy`. And yes, this should work is such case.

Reply:
@fkanout Sounds like you would prefer a type guard? Which is sort of what this function is.

Reply:
I realized after I typed that comment we actually have type guards for all of these checks and the `isMetricRate` and `isInterfaceRate` are really a composition of these type checks.